icu4c/source/test/cintltst/cloctst.c

```diff
@@ -704,9 +704,19 @@ static void TestDisplayNames()
     /* test that the default locale has a display name for its own language */
     errorCode=U_ZERO_ERROR;
     length=uloc_getDisplayLanguage(NULL, NULL, buffer, UPRV_LENGTHOF(buffer), &errorCode);
-    if(U_FAILURE(errorCode) || (length<=3 && buffer[0]<=0x7f)) {
-        /* check <=3 to reject getting the language code as a display name */
-        log_data_err("unable to get a display string for the language of the default locale - %s (Are you missing data?)\n", u_errorName(errorCode));
+    /* check <=3 to reject getting the language code as a display name */
+    if(U_FAILURE(errorCode) || (length<=3 && buffer[0]<=0x7f && buffer[1]<=0x7f)) {
+        char locale[3] = {(char)(buffer[0]), (char)(buffer[1]), '\0'};
```

Comment:
Please use a UTF-16 --> default charset conversion function to make this portable. (Check one of your recent PRs.)

Also, if length==3 then this code would drop the third character.

Also, if U_FAILURE() then the buffer might not contain anything and we shouldn't read it at all.